### PR TITLE
Implement TOC

### PR DESCRIPTION
**Pull request by tuchfarber**
Features:
 - expand/collapse
 - link to page
 - highlight current page
 - auto expand to current page

### CODE REVIEW

**Comment by bill-filler** [comment]
Regarding highlighting. Current behavior highlights the current page which is good. But then on hover we also highlight the item under the mouse and then have two highlighted items that are the same highlight color. It's a bit weird looking to me, especially if one is directly beneath the other, but it might just be me :) One possibility would be to slightly change the color of the hover highlight so it's clear it's different. Or remove the highlight from the current page while hover over another and revert back once the hover is complete.

File: src/components/TOCViewer/index.jsx [hunk 1]
```diff
@@ -1,32 +1,147 @@
...
+    />
   ))
 );
 
 const TOCViewer = props => (
   // If the title is an empty string do not display one
-  <div className="toc-border">
-    {props.journal.title.trim() && <div className="journal-title">{props.journal.title}</div>}
+  <div className="toc">
+    <div className="toc-title">TABLE OF CONTENTS</div>
```
Comment (bill-filler) [inline src/components/TOCViewer/index.jsx]: general question here about labels: what is our strategy for i18n of strings? In django templates we have the _() get text method, guessing we need equivalent for React.

**Comment by bill-filler** [comment]
Would be good to have a plan for structure updates, i.e. if a new page is published or unpublished how and when are we going to refresh the tree? doesn't have to be in this PR but if not please create a story so we can do it after

**Comment by bill-filler** [comment]
Can you update the "Menu" button in this PR to look more like the icon in the design?

File: src/components/TOCViewer/index.jsx (hunk 1 above)
Comment (tuchfarber) [reply src/components/TOCViewer/index.jsx]: I know it's been discussed, but I don't know where it stands. I can ask in FedX on Thursday and if we have a plan in place, I can add a story to get i18n working in the app on our backlog.

**Pull request merged**